internal/worker/controlsocket/worker.go

```diff
@@ -157,23 +215,30 @@ func (w *Worker) handleAddMetricsUser(resp http.ResponseWriter, req *http.Reques
 	w.writeResponse(resp, code, infof("created user %q", parsedBody.Username))
 }
 
-func (w *Worker) addMetricsUser(username, password string) (int, error) {
+func (w *Worker) addMetricsUser(ctx context.Context, username string, password auth.Password) (int, error) {
 	err := validateMetricsUsername(username)
 	if err != nil {
 		return http.StatusBadRequest, err
 	}
 
-	if password == "" {
-		return http.StatusBadRequest, errors.NotValidf("empty password")
+	creatorUser, err := w.userService.GetUserByName(ctx, userCreator)
```

Comment:
Can this not be done outside the handler and have the creatorUser.UUID stored in the worker? Its getting the same user every time based on a constant string and only using its UUID.

Reply:
I think this is ok for now, but you're right. Maybe revisit this in the future.